samtranslator/model/sam_resources.py

```diff
@@ -874,18 +877,25 @@ def _construct_version(
         #                 and next hashes. The chances that two subsequent hashes collide is fairly low.
         prefix = "{id}Version".format(id=self.logical_id)
         logical_dict = {}
-        try:
-            logical_dict = code_dict.copy()
-        except (AttributeError, UnboundLocalError):
-            pass  # noqa: try-except-pass
+        # We can't directly change AutoPublishAlias as that would be a breaking change, so we have to add this opt-in
+        # property that when set to true would change the lambda version whenever a property in the lambda function changes
+        if self.AutoPublishAliasAllProperties:
+            properties = function._generate_resource_dict().get("Properties", {})
+            logical_dict = properties
         else:
-            if function.Environment:
-                logical_dict.update(function.Environment)
-            if function.MemorySize:
-                logical_dict.update({"MemorySize": function.MemorySize})
-            # If SnapStart is enabled we want to publish a new version, to have the corresponding snapshot
-            if function.SnapStart and function.SnapStart.get("ApplyOn", "None") != "None":
-                logical_dict.update({"SnapStart": function.SnapStart})
+            try:
+                logical_dict = code_dict.copy()
+            except (AttributeError, UnboundLocalError):
+                pass  # noqa: try-except-pass
```

Comment:
We should use https://docs.python.org/3/library/contextlib.html#contextlib.suppress to be more specific. Also some comment can help to understand why